code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthTokenDaoImpl.java

```diff
@@ -0,0 +1,84 @@
+package io.cattle.platform.iaas.api.auth.dao.impl;
+
+import static io.cattle.platform.core.model.tables.AuthTokenTable.AUTH_TOKEN;
+
+import io.cattle.platform.core.constants.CommonStatesConstants;
+import io.cattle.platform.core.dao.GenericResourceDao;
+import io.cattle.platform.core.model.AuthToken;
+import io.cattle.platform.db.jooq.dao.impl.AbstractJooqDao;
+import io.cattle.platform.iaas.api.auth.SecurityConstants;
+import io.cattle.platform.iaas.api.auth.dao.AuthTokenDao;
+import io.cattle.platform.iaas.api.auth.integration.github.GithubConstants;
+import io.cattle.platform.iaas.api.filter.apikey.ApiKeyFilter;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.process.ObjectProcessManager;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+
+public class AuthTokenDaoImpl extends AbstractJooqDao implements AuthTokenDao{
+
+    @Inject
+    GenericResourceDao resourceDao;
+    @Inject
+    ObjectManager objectManager;
+    @Inject
+    ObjectProcessManager objectProcessManager;
+
+    @Override
+    public AuthToken getTokenById(String key) {
+        return create()
+                .selectFrom(AUTH_TOKEN)
+                .where(AUTH_TOKEN.STATE.eq(CommonStatesConstants.ACTIVE)
+                        .and(AUTH_TOKEN.KEY.eq(key))
+                        .and(AUTH_TOKEN.VERSION.eq(SecurityConstants.TOKEN_VERSION))
+                        .and(AUTH_TOKEN.REMOVE_TIME.greaterThan(new Date())))
+                                .orderBy(AUTH_TOKEN.ID.asc()).fetchOneInto(AuthToken.class);
+    }
+
+    @Override
+    public AuthToken createToken(String jwt, String provider) {
+        Map<Object, Object> properties = new HashMap<>();
+        if (StringUtils.isBlank(jwt)){
+            throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR, "NoJwtToSave", "Cannot save a null jwt.",
+                    null);
+        }
+        properties.put(AUTH_TOKEN.VALUE, jwt);
+        properties.put(AUTH_TOKEN.KEY, ApiKeyFilter.generateKeys()[1]);
+        properties.put(AUTH_TOKEN.VERSION, SecurityConstants.TOKEN_VERSION);
+        properties.put(AUTH_TOKEN.PROVIDER, provider);
+        Date expiry = new Date(System.currentTimeMillis() + SecurityConstants.TOKEN_EXPIRY_MILLIS.get());
+        properties.put(AUTH_TOKEN.REMOVE_TIME, expiry);
```

Comment:
I believe we agreed on a different field name then remove_time.  @cjellick Do you have notes?